apisix/plugins/ldap-auth.lua

```diff
@@ -0,0 +1,171 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core = require("apisix.core")
+local ngx = ngx
+local ngx_re = require("ngx.re")
+local ipairs = ipairs
+local consumer_mod = require("apisix.consumer")
+local lualdap = require "lualdap"
+
+local lrucache = core.lrucache.new({
+    ttl = 300, count = 512
+})
+local consumers_lrucache = core.lrucache.new({
+    type = "plugin",
+})
+
+local schema = {
+    type = "object",
+    title = "work with route or service object",
+    properties = {
+        basedn = { type = "string" },
+        ldapuri = { type = "string" },
+        usetls = { type = "boolean" },
```

Comment:
Would you use a snake case for the field name, like `base_dn` / `ldap_uri`?

Reply:
Ok change